packages/quantic/force-app/main/default/lwc/quanticTabBar/e2e/quanticTabBar.e2e.ts

```diff
@@ -0,0 +1,199 @@
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+const extraSmallViewportWidth = 200;
+const smallViewportWidth = 260;
+const mediumViewportWidth = 300;
+const standardViewportHeight = 1080;
+
+const viewportWidths = [
+  {
+    label: 'extra small',
+    width: extraSmallViewportWidth,
+    primaryExpectation: 'the selected tab ',
+    secondaryExpectation: 'and the more button should be shrunk',
+    numberOfTabsDisplayed: 1,
+    numberOfOptionsInDropdown: 3,
+    moreButtonLabel: '',
+  },
+  {
+    label: 'small',
+    width: smallViewportWidth,
+    primaryExpectation: 'the selected tab',
+    secondaryExpectation: '',
+    numberOfTabsDisplayed: 1,
+    numberOfOptionsInDropdown: 3,
+    moreButtonLabel: 'More',
+  },
+  {
+    label: 'medium',
+    width: mediumViewportWidth,
+    primaryExpectation: 'the first two tabs',
+    secondaryExpectation: '',
+    numberOfTabsDisplayed: 2,
+    numberOfOptionsInDropdown: 2,
+    moreButtonLabel: 'More',
+  },
+];
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+  test.describe(`quantic tab bar ${useCase.label}`, () => {
+    test.describe("when the container's width can fit all the tabs", () => {
+      test('should display all the tabs without displaying the dropdown list', async ({
+        tabBar,
+      }) => {
+        const expectedNumberOfTabs = 4;
+        const tabs = tabBar.allVisibleTabs;
+
+        expect(tabs).not.toBeNull();
+        expect(await tabs.count()).toEqual(expectedNumberOfTabs);
+
+        const isMoreDropdownVisible = await tabBar.dropdown.isVisible();
+        expect(isMoreDropdownVisible).toBe(false);
+      });
+    });
+
+    viewportWidths.forEach(
+      ({
+        label,
+        width,
+        primaryExpectation,
+        secondaryExpectation,
+        numberOfTabsDisplayed,
+        numberOfOptionsInDropdown,
+        moreButtonLabel,
+      }) => {
+        test.describe(`when the viewport is resized to a ${label} width`, () => {
+          test.use({
+            viewport: {width: width, height: standardViewportHeight},
+          });
+          test(`should display only ${primaryExpectation}, the other tabs should be displayed in the dropdown list ${secondaryExpectation}`, async ({
+            tabBar,
+          }) => {
+            const displayedTabs = tabBar.allVisibleTabs;
+
+            expect(displayedTabs).not.toBeNull();
+            expect(await displayedTabs.count()).toEqual(numberOfTabsDisplayed);
+
+            const isMoreDropdownVisible = await tabBar.dropdown.isVisible();
+            expect(isMoreDropdownVisible).toBe(true);
+
+            await tabBar.clickMoreButton();
+
+            const dropdownOptions = tabBar.allDropdownOptions;
+            expect(dropdownOptions).not.toBeNull();
+            expect(await dropdownOptions.count()).toEqual(
+              numberOfOptionsInDropdown
+            );
```

Comment:
"Options" is fairly generic, I think it's clearer to specify these are tabs.

```suggestion
            const tabsInDropdown = tabBar.allTabsInDropdown;
            expect(tabsInDropdown).not.toBeNull();
            expect(await tabsInDropdown.count()).toEqual(
              numberOfTabsInDropdown
            );
```